GitUI/WindowsJumpListManager.cs

```diff
@@ -18,24 +19,31 @@ namespace GitUI
     /// <seealso href="https://www.sevenforums.com/news/44368-developing-windows-7-taskbar-thumbnail-toolbars.html" />
     /// <seealso href="https://github.com/jlnewton87/Programming/blob/master/C%23/Windows%20API%20Code%20Pack%201.1/source/WindowsAPICodePack/Shell/Taskbar/JumpList.cs" />
     /// <inheritdoc />
+    [Export(typeof(WindowsJumpListManager))]
     public sealed class WindowsJumpListManager : IDisposable
     {
+        private readonly IRepositoryDescriptionProvider _repositoryDescriptionProvider;
         private ThumbnailToolBarButton? _commitButton;
         private ThumbnailToolBarButton? _pushButton;
         private ThumbnailToolBarButton? _pullButton;
         private string? _deferredAddToRecent;
+
         private bool ToolbarButtonsCreated => _commitButton is not null;
-        private readonly IRepositoryDescriptionProvider _repositoryDescriptionProvider;
 
-        public WindowsJumpListManager(IRepositoryDescriptionProvider repositoryDescriptionProvider)
+        static WindowsJumpListManager()
         {
-            _repositoryDescriptionProvider = repositoryDescriptionProvider;
             if (TaskbarManager.IsPlatformSupported)
             {
                 TaskbarManager.Instance.ApplicationId = AppSettings.ApplicationId;
             }
         }
 
+        [ImportingConstructor]
+        public WindowsJumpListManager(IRepositoryDescriptionProvider repositoryDescriptionProvider)
+        {
+            _repositoryDescriptionProvider = repositoryDescriptionProvider;
+        }
+
         ~WindowsJumpListManager()
```

Comment:
💡 MEF will dispose instantiated parts when the container is disposed. There is no need to include a finalizer here.